src/buildercore/lifecycle.py

```diff
@@ -42,12 +42,12 @@ def push(lst, rec):
     if not lst or lst[-1] != rec:
         lst.append(rec)
 
-def restart(stackname):
+def restart(stackname, initial_states='pending|running|stopping|stopped'):
```

Comment:
why not an actual data structure (list? set?) here rather than a `|`-separated string?

Reply:
it's how it's handled higher up in buildercore. I dimly recall it having something to do with boto2 but could be mistaken. I'm not going to change it for this PR but I think there are lots of little places where the libraries have improved that we could revisit and trim some code.